src/RL/Program.hs

```diff
@@ -65,8 +65,25 @@ getExitName = name . head . filter isExit
 getExitLabel :: [Block a b] -> a
 getExitLabel = label . head . filter isExit
 
-getEntryProcedure :: [Procedure a b] -> EM (Procedure a b)
-getEntryProcedure p =
+getMainProcedure :: [Procedure a b] -> EM (Procedure a b)
+getMainProcedure p =
```

Comment:
There is merit in having it due to it helping future-proof the code in case we refactor the AST again, but I don't think it will ever make sense for the program to fail to find the main procedure (so not monadic).